models.py

```diff
@@ -28,10 +31,16 @@ def generate_compiled_segmentation_model(model_name, model_parameters, num_class
                                          optimizing_class_id=None, optimizing_input_threshold=None,
                                          optimized_class_thresholds=None):
 
+    # alter input_shape due to inability of yaml to accept tupples!
+    if 'input_shape' in model_parameters:
+        model_parameters['input_shape'] = tuple(model_parameters['input_shape'])
+    else:  # to guarantee the old config files still work
+        model_parameters['input_shape'] = (None, None, 1)
+
+    # this is the only optimizer currently in use
     # These are the only model, loss, and optimizer currently supported
-    assert model_name == 'Unet'
-    assert loss == 'cross_entropy'
     assert optimizer == 'adam'
+    assert loss == 'cross_entropy'
```

Comment:
can we get rid of these asserts? @rak5216 do you still want this to break if different optimizers and loss functions are selected?